File: Warm-up/Running Sum of 1D Array/java/Solution.java
class Solution {
    public int[] runningSum(int[] nums) {
        if (nums == null || nums.length == 0) {
            return nums; // Return as is
        }

        for (int i = 1; i < nums.length; i++) {
            nums[i] += nums[i - 1];
        }

        return nums;
    }

    // Main method to test the runningSum method
    public static void main(String[] args) {
        Solution solution = new Solution();

        int[] nums = { 1, 2, 3, 4 };
        int[] result = solution.runningSum(nums);

        // Print the running sum
        System.out.print("Running Sum: ");
        for (int num : result) {
            System.out.print(num + " ");
        }
    }
}

File: Warm-up/Running Sum of 1D Array/go/Solution.go
package main

import "fmt"

func runningSum(nums []int) []int {
	for i := 1; i < len(nums); i++ {
		nums[i] += nums[i-1]
	}
	return nums
}

func main() {
	nums := []int{1, 2, 3, 4}
	result := runningSum(nums)
	fmt.Println(result) // Output: [1, 3, 6, 10]
}
